plugins/com.google.cloud.tools.eclipse.appengine.localserver/src/com/google/cloud/tools/eclipse/appengine/localserver/ui/GcpLocalRunTab.java

```diff
@@ -234,24 +231,30 @@ public void modifyText(ModifyEvent event) {
 
     createServiceKey = new Button(composite, SWT.NONE);
     createServiceKey.setText(Messages.getString("create.new.service.key")); //$NON-NLS-1$
+    createServiceKey.setToolTipText(
+        Messages.getString("create.new.service.key.tooltip")); //$NON-NLS-1$
     createServiceKey.addSelectionListener(new SelectionAdapter() {
       @Override
       public void widgetSelected(SelectionEvent event) {
-        createServiceAccountKey(getServiceAccountKeyPath());
+        BusyIndicator.showWhile(createServiceKey.getDisplay(), new Runnable() {
+          @Override
+          public void run() {
+            createServiceAccountKey(getServiceAccountKeyPath());
```

Comment:
What if they've provided a path in the text?  Should we use it?

Reply:
We'll need a slightly different UI if we do that. We may think about it later.